database-commons/src/main/java/io/cdap/plugin/db/connector/DBSpecificPath.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright © 2021 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package io.cdap.plugin.db.connector;
+
+import io.cdap.plugin.common.db.DBConnectorPath;
+
+import javax.annotation.Nullable;
+
+/**
+ * DB specific Path that parses the path in the request of connection service
+ * A valid path can start with/without a slash, e.g. "/" , "", "/database/schema/table", "database/schema/table" are
+ * all valid.
+ * A valid path can end with/without a slash, e.g. "/", "", "/database/schema/table/", "database/schema/table/" are all
+ * valid.
+ * A valid path should contain at most two parts for those DB that don't support schema (e.g. mysql) or three parts
+ * for those DB that support schema (e.g. postgresql, oracle, sqlserver) separated by slash. The first part is
+ * database name. The second part is schema name for those DB that support schema or table name for those DB that
+ * don't support schema. Third part is table name (only applicable for those DB that support schema). Any part is
+ * optional. e.g. "/" , "/database" , "/database/schema" (for those DB that support Schema), "database/schema/table"
+ * (for those DB that support Schema), "/database/table" (for those DB that don't support Schema)
+ * Consecutive slashes are not valid , it will be parsed as there is an empty string part between the slashes. e.g.
+ * "//a" will be parsed as database name as empty and schema name or table name as "a". Similarly "/a//" will be parsed
+ * as database name as "a" and table name or schema name as empty.
+ *
+ */
+public class DBSpecificPath implements DBConnectorPath {
+  private final boolean supportSchema;
+  private String database;
+  private String schema;
+  private String table;
+
+  public DBSpecificPath(String path, boolean supportSchema) {
```

Comment:
Consider making a factory method. This would allow you to have all fields final

Reply:
good idea, done :)